### PR TITLE
G dialog

### PR DESCRIPTION
**Pull request by SonlamBKIC**

### CODE REVIEW

File: src/components/GDialog/GDialog.vue [hunk 1]
```diff
@@ -1,40 +1,181 @@
...
-        return {
-          'dialog-wrapper__light': !this.dark,
-          'dialog-wrapper__dark': this.dark,
+		setup (props, context) {
+      const { model: isActive } = getVModel(props, context);
+      const { attachToRoot, attachToParent } = detachable(props, context);
+      const { getMaxZIndex } = stackable(props, context);
+
+      // Stacking
+      const zIndex = computed(() => {
```
Comment (longnh301) [inline src/components/GDialog/GDialog.vue]: day la zIndex cua cai gi?

File: src/components/GDialog/GDialog.vue [hunk 2]
```diff
@@ -1,40 +1,181 @@
...
+			const renderOverlay = computed(() => !props.hideOverlay && !props.fullscreen && lazyRender.value);
+
+
+
+      // Lazy/Eager
+      // TODO: convert to bootable mixin
+      const isBooted = reactive({
+        value: false
+      });
+      const lazyRender = computed(() => isBooted.value || !props.lazy);
```
Comment (longnh301) [inline src/components/GDialog/GDialog.vue]: lazyRender -> renderContent

Comment (SonlamBKIC) [reply src/components/GDialog/GDialog.vue]: em sua roi day a

File: src/components/GDialog/GDialog.vue [hunk 3]
```diff
@@ -1,40 +1,181 @@
...
+      };
+			const directiveArgs = {
+        closeConditional,
+				include: () => []
+      };
+
+			// Press ESC key
+			const tabIndex = computed(() => isActive.value ? 0 : undefined);
+
+			function onKeydown(e) {
```
Comment (longnh301) [inline src/components/GDialog/GDialog.vue]: group code according to usage

File: src/mixins/stackable.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { getZIndex } from '../utils/helpers';
+import { ref } from '@vue/composition-api'
+
+export default function stackable(props, context) {
+  const stackMinZIndex = ref(200);
+
+  function getMaxZIndex(exclude) {
+    //const base = context.refs.el;
+
+    const zis = [stackMinZIndex.value];
```
Comment (longnh301) [inline src/mixins/stackable.js]: zis -> activeElementsZIndexes

File: src/mixins/detachable.js [hunk 5]
```diff
@@ -1,21 +1,60 @@
 import { onMounted } from '@vue/composition-api';
```
Comment (longnh301) [inline src/mixins/detachable.js]: comment: // requires template refs: activator, content, el

File: src/components/GDialog/GDialog.vue [hunk 6]
```diff
@@ -1,41 +1,206 @@
...
+        const clickedInsideContent = context.refs.content.contains(e.target);
+				if (clickedInsideContent) return false;
+
+				// If z-index of current element is lower than the current active z-index then do not close when click outside
+        return getZIndex(context.refs.wrapper) >= getMaxZIndex(context.refs.wrapper);
+      };
+
+      const clickOutsideDirective = {
+				value: () => {
+					if (!props.persistent) {
```
Comment (longnh301) [inline src/components/GDialog/GDialog.vue]: move to closeConditional()

**Pull request merged**